cylc/flow/cfgspec/globalcfg.py

```diff
@@ -17,7 +17,9 @@
 
 import os
 from typing import List, Optional, Tuple
-import packaging.version
+
+from pkg_resources import parse_version
+from pkg_resources.extern.packaging.version import Version
```

Comment:
VSCode can't resolve this and I don't think mypy can either (I tried accessing a non-existent attribute of `smart_ver` and mypy didn't complain. So might as well do this
```suggestion
from typing import Any
```

Reply:
> VSCode can't resolve this

Eh?

If the import wasn't resolvable Python would crash with traceback:

```console
$ python
Type "help", "copyright", "credits" or "license" for more information.
>>> from pkg_resources.extern.packaging.version import Version
>>> Version
<class 'pkg_resources.extern.packaging.version.Version'> 
```

Reply:
```diff
diff --git a/cylc/flow/cfgspec/globalcfg.py b/cylc/flow/cfgspec/globalcfg.py
index 960e1affd..c11c29db0 100644
--- a/cylc/flow/cfgspec/globalcfg.py
+++ b/cylc/flow/cfgspec/globalcfg.py
@@ -770,7 +770,7 @@ def get_version_hierarchy(version: str) -> List[str]:
 
     """
     smart_ver: Version = parse_version(version)
-    base = [str(i) for i in smart_ver.release]
+    base = [str(i) for i in smart_ver.asdf]
     hierarchy = ['']
     hierarchy += ['.'.join(base[:i]) for i in range(1, len(base) + 1)]
     if smart_ver.pre:  # alpha/beta (excluding dev) part of version
```
followed by
```console
$ mypy --explicit-package-bases 
Success: no issues found in 182 source files
```
But if I change it to a `packaging.version.Version` then I get as expected
```console
$ mypy --explicit-package-bases 
cylc/flow/cfgspec/globalcfg.py:773: error: "Version" has no attribute "asdf"
Found 1 error in 1 file (checked 182 source files)
```

Reply:
Doesn't that just mean that setuptools have not added type hints to their code yet?

Reply:
Had a look at `~/miniconda3/envs/cylc8/lib/python3.7/site-packages/pkg_resources/extern/__init__.py`, not seen this kind of thing before. I'm guessing mypy can't follow this

https://github.com/pypa/setuptools/blob/ee6a5ffd1b3ec686339e00713542a530dbb65bc8/pkg_resources/extern/__init__.py

Reply:
Swapped to this which seems to be the thing it is "vendoring"?

```diff
-from pkg_resources.extern.packaging.version import Version
+from pkg_resources._vendor.packaging.version import Version
```